crates/dojo/bindgen/src/plugins/unrealengine/mod.rs

```diff
@@ -0,0 +1,1376 @@
+            "bytes31" => "FString".to_string(),
+            "ClassHash" => "FString".to_string(),
+            "ContractAddress" => "FString".to_string(),
+            "ByteArray" => "FString".to_string(),
+            "Option" => {
+                // TODO: TOptional is not handled in blueprints so we can't really use that
+                panic!("Option type not handled");
+            }
+            "array" => {
+                if let Token::Array(array) = token {
+                    let mapped = UnrealEnginePlugin::map_type(&array.inner);
+                    format!("TArray<{}>", mapped)
+                } else {
+                    panic!("Invalid array token: {:?}", token);
+                }
+            }
+            "tuple" => {
+                panic!("Tuple type not handled");
+            }
+            "generic_arg" => {
+                if let Token::GenericArg(g) = &token {
+                    g.clone()
+                } else {
+                    panic!("Invalid generic arg token: {:?}", token);
+                }
+            }
+            _ => {
+                let mut type_name = token.type_name().to_string();
+
+                if let Token::Composite(composite) = token {
+                    if !composite.generic_args.is_empty() {
+                        type_name += &format!(
+                            "<{}>",
+                            composite
+                                .generic_args
+                                .iter()
+                                .map(|(_, t)| UnrealEnginePlugin::map_type(t))
+                                .collect::<Vec<_>>()
+                                .join(", ")
+                        )
+                    }
+
+                    let type_str = match composite.r#type {
+                        CompositeType::Struct => "F".to_owned() + &type_name,
+                        CompositeType::Enum => "ED".to_owned() + &type_name,
+                        _ => type_name,
+                    };
+                    return type_str;
+                }
+                type_name
+            }
+        }
+    }
+
+    fn generated_header() -> String {
+        format!(
+            "// Generated by dojo-bindgen on {}. Do not modify this file manually.\n",
+            chrono::Utc::now().to_rfc2822()
+        )
+    }
+
+    fn header_imports() -> String {
+        "
+#pragma once
+
+#include \"CoreMinimal.h\"
+#include \"GameFramework/Actor.h\"
+#include \"DojoModule.h\"
+#include \"Account.h\"
+#include \"GeneratedHelpers.generated.h\"
+
+UCLASS(BlueprintType)
+class UDojoModel : public UObject
+{
+    GENERATED_BODY()
+
+public:
+    UPROPERTY(EditAnywhere, BlueprintReadWrite)
+    FString DojoModelType;
+};
+
+    "
+        .to_string()
+    }
+
+    fn cppfile_imports() -> String {
+        "
+#include \"GeneratedHelpers.h\"
+#include <string>
+#include <iomanip>
+#include <sstream>
+#include <memory>
+#include \"Async/Async.h\"
+
+"
```

Comment:
```suggestion
        r#"
#include "GeneratedHelpers.h"
#include <string>
#include <iomanip>
#include <sstream>
#include <memory>
#include "Async/Async.h"

"#
```
Favorise the use of raw string for easier maintenance of the generated code.